crawl-ref/source/fight.cc

```diff
@@ -751,7 +752,8 @@ void attack_cleave_targets(actor &attacker, list<actor*> &targets,
     while (attacker.alive() && !targets.empty())
     {
         actor* def = targets.front();
-        if (def && def->alive() && !_dont_harm(attacker, *def) && cleave_target_adjacent(attacker, *def))
+        bool def_adjacent = adjacent(attacker.pos(), def->pos());
```

Comment:
I didn't consider that def could be null; 

Something like this will take that into account:
        if (def && def->alive() && !_dont_harm(attacker, *def) && adjacent(attacker.pos(), def->pos()))